recipes/pro-mdnsd/all/patches/fix_signed_overfload_optimization.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/libmdnsd/sdtxt.c b/libmdnsd/sdtxt.c
+index c18c852..67e59bc 100644
+--- a/libmdnsd/sdtxt.c
++++ b/libmdnsd/sdtxt.c
+@@ -61,7 +61,7 @@ unsigned char *sd2txt(xht_t *h, int *len)
+ 	return raw;
+ }
+ 
+-xht_t *txt2sd(const unsigned char *txt, int len)
```

Comment:
I'm intrigued what error this is fixing, because the int is passed by value.

Reply:
it is a weird GCC warning for signed integer overflow assert, which is sort of explained at [stackoverflow](https://stackoverflow.com/questions/12984861/dont-understand-assuming-signed-overflow-warning). 

In all honesty, I don't find the for loop implementation to be easily understandable, but it works. If it causes problems in the future, then we can open an issue on the project itself

Finally I didn't want to remove the `Werror` flag to solve this error or change overflow assertions, since I want the CI to fail if any warning that is not being handled pops up

Reply:
Thanks for the links!
I managed to reproduce it on godbolt: https://godbolt.org/z/GRypL2
By switching the conditionals in the for loop conditional, the error disappears: https://godbolt.org/z/IpPFsX

@Pro

Reply:
Thank @madebr! I changed the [fix_signed_overfload_optimization.patch](https://github.com/Hahn-Schickard/conan-center-index/blob/mdnsd/recipes/pro-mdnsd/all/patches/fix_signed_overfload_optimization.patch) to apply your suggestion, while @Pro implements it

Reply:
I created a PR to apply two of your patches (install and signed overload):

https://github.com/Pro/mdnsd/pull/18

As soon as my CI runs trough, I will create a 0.8.1 release. Then you can remove the patches here and we avoid possible inconsistencies.

Btw I can also do the same for open62541 (as long as the patches don't fiddle too much with the base), as I'm one of the core devs there.

Reply:
Done: https://github.com/Pro/mdnsd/releases/tag/v0.8.1

Reply:
Hey @Pro, I think that the install targets patch will add a new includes directory (mdnsd) inside of with open62541 dep package.

It shouldn't cause any problems, however it is something to consider for the package maintainers. 

I'll need to look into how I can include this recipe as a dependency for open62541 project and that might require me to do slight patches to the root `CMakeLists.txt` for open62541, but I'll do that after we have a working package

Reply:
I will check that with the update to the new mdnsd release. I'm also a core dev at open62541

Reply:
sure, is it safe to rebase the recipe on 0.8.2?

Reply:
I will check if open62541 (master) still builds with the changes in 0.8.2 and update the PR https://github.com/open62541/open62541/pull/3740 correspondingly

Reply:
@Pro, there is a CMake target install error, I explained it in [your PR](https://github.com/Pro/mdnsd/pull/19) after you merged it. I can patch it for conan, but it will cause problems for open62541 deb packaging

Reply:
Can you point me to the corresponding explanation / error and maybe create a PR directly in mdnsd repo?

Reply:
I'll make a direct PR, it will be easier

Reply:
@saukijan FYI, updating mdnsd to 0.8.3 which includes the CMake fixes does not break functionality in open62541 (master and 1.0).
Just created the PR in open62541 and all CI (including unit tests for mdnsd functionality) run through. https://github.com/open62541/open62541/pull/3740

Reply:
Thanks for the update @Pro, but does the CI try to package the binaries and do a test install? That's where problems might show up.